x-pack/plugin/core/src/test/java/org/elasticsearch/license/internal/TrialLicenseEraTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.license.internal;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class TrialLicenseEraTests extends ESTestCase {
+
+    public void testCanParseAllVersions() {
+        for (var version : Version.getDeclaredVersions(Version.class)) {
+            TrialLicenseEra era = TrialLicenseEra.fromString(version.toString());
+            assertThat((byte) era.asInt(), equalTo(version.major));
+        }
+    }
+
+    public void testVersionWireCompatibility() throws IOException {
+        for (var version : Version.getDeclaredVersions(Version.class)) {
+            versionToEraSerialization(version);
+        }
+
+        for (int i = 2; i < 100; i++) { // old Version class only grows to 99, so that's the limit
+            eraToVersionSerialization(new TrialLicenseEra(i));
+        }
+    }
+
+    // This simulates a node of a version that supports transport eras receiving a message which is still using old-school Versions
+    private TrialLicenseEra versionToEraSerialization(Version version) throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            output.setTransportVersion(TransportVersion.current());
```

Comment:
Should the test also include Transport version prior to the introduction of `TransportVersion` ?

Reply:
I don't think that using a different version there changes anything in this test; being a unit test this can't really-actually check BWC so all it's really doing is feeding the output of `writeTo` of one into the `StreamInput` constructor of the other. If I'm wrong and there's some logic we should make sure is tested here please point me to it; this is an unorthodox thing I'm doing and I want to make sure all the bases are covered.